server/routes/index.js

```diff
@@ -1,8 +1,120 @@
 const express = require("express");
 const router = express.Router();
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const auth = require("../middleware/auth");
+const User = require("../models/user");
 
-router.get("/welcome", function (req, res, next) {
+// testing root route
+router.get("/", function (req, res, next) {
   res.status(200).send({ welcomeMessage: "Step 1 (completed)" });
 });
 
+// register route
+router.post("/register", async (req, res) => {
+  const { email, confirmEmail, password, confirmPassword } = req.body;
+
+  try {
+    // validate if all fields are filled
+    if (!email || !confirmEmail || !password || !confirmPassword) {
+      return res.status(400).send("All fields need to be filled!");
+    }
+    // validate if email aleady exist
+    const userExist = await User.findOne({ email }).exec();
+    if (userExist) {
+      return res.status(409).send("Email already exist!");
+    }
+    // validate if password length is lesser than 6
+    if (password.length < 6) {
+      return res
+        .status(401)
+        .send("Password needs to be at least 6 characters long");
+    }
+    // validate confirm email and confirm password
+    if (email !== confirmEmail || password !== confirmPassword) {
+      return res
+        .status(401)
+        .send(
+          "Email and confirm Email or Password and confirm Password do not match"
+        );
+    }
+    // hash password - bcrypt
+    const salt = await bcrypt.genSalt();
+    const hashedPassword = await bcrypt.hash(password, salt);
+    // create new user
+    const newUser = new User({
+      email,
+      password: hashedPassword,
+    });
```

Comment:
I think we need first name and last name right (or a name field)

Reply:
Do we need to recheck all input data if it was checked on the frontend? Following the mockups to register new user we should have 3 inputs: Username, email, password
<img width="1680" alt="Screenshot 2020-07-21 at 16 03 56" src="https://user-images.githubusercontent.com/22850021/88241289-b048bf00-cc57-11ea-92ba-fe1f58fbd29c.png">